### PR TITLE
Make CI work!

### LINKED ISSUE #94: Fix CI
CI is out of date. See https://github.com/jpvillaisaza/slack-web/pull/93#issuecomment-1233516085 for a recommendation to use `haskell-ci` or switch to GitHub Actions.

### PR DESCRIPTION
**Pull request by lf-**
This uses GitHub Actions and haskell-ci. I decided not to figure out CircleCI.

You do need the latest haskell-ci version from git to be able to generate for GHC 9.4.2. But it works!

This also builds and tests with GHC 9.4.2 ;-)

It also comes with a bonus Nix flake, because I can't help myself I guess 😆 

Fixes #94

### CODE REVIEW

**Comment by lf-** [comment]
The repo configuration needs to be fixed in some way, it seems. GitHub actions does pass on my branch at least.

**Review by jpvillaisaza — APPROVED** [review]

**Comment by jpvillaisaza** [comment]
@lf- Maybe after merging it starts running. I made a small change to the repo configuration so that actions are enabled for forks without requiring an approval (in most cases), but didn't find anything else. I also disabled the CircleCI checks.

Let me know if this is okay to merge now and I can go ahead and do that to see if it starts running.

Thanks!

**Pull request closed without merging**
